sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -299,6 +300,34 @@ public Instant getCurrentTimestamp() throws NoSuchElementException {
     return curTimestamp;
   }
 
+  private static final byte[] EMPTY_RECORD_ID = new byte[0];
+
+  @Override
+  public byte[] getCurrentRecordId() throws NoSuchElementException {
```

Comment:
the super impl throws an exception if getCurrentSource().requiresDeduping() is requested and this isn't implemented.

Maybe it woudl be better to delegate to super if !offsetBasedDeduplicationSupported so it fails more loudly if that gets set in the future?